var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -136,24 +170,21 @@ class Llvm(CMakePackage, CudaPackage):
     )
     variant(
         "llvm_dylib",
-        default=False,
+        default=False, when='~shared_libs',
```

Comment:
I think it's better to do it the other way around. `shared_libs` is the exception.

Reply:
Yes, shared_libs should almost never be used, where llvm_dylib is the supported option.

Reply:
So, the shared component libs are not particularly well supported by llvm anyways and nobody actually uses that configuration. @haampie Can we just drop the `shared_libs` and `link_llvm_dylib` variants entirely and just have `llvm_dylib`?  This would have the package always building the static component libs and optionally the shared combined lib.

Reply:
No, at least not `link_llvm_dylib`. That is a commonly used configuration for reducing total space required (by as much as 15gb in debug builds in some cases).  It's the normal way most distros link llvm, and we're not getting rid of it.

I actually would be totally fine with removing `shared_libs`, but because of how widely used and depended on the llvm package is, we probably need to go through a deprecation cycle to actually do that, so if you want to feel free to make it deprecated now and file a reminder issue for the release after next to actually remove it.

Reply:
Or 550MB of `llvm-*` executables vs 53MB, it's quite significant

Reply:
I't significant for sure.  I mean removing the `link_llvm_dylib` variant but still use the shared llvm dylib if it's built.  i.e:
```
        cmake_args.extend([
            from_variant("BUILD_SHARED_LIBS", "shard_libs"),
            from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"),
            from_variant("LLVM_LINK_LLVM_DYLIB", "link_llvm_dylib"),
```
becomes
```
        cmake_args.extend([
            define("BUILD_SHARED_LIBS", False),
            from_variant("LLVM_BUILD_LLVM_DYLIB", "llvm_dylib"),
            from_variant("LLVM_LINK_LLVM_DYLIB", "llvm_dylib"),
```
I just don't really see the use case for `+llvm_dylib ~link_llvm_dylib`.  Am I missing something (which I certainly could be)

Reply:
That's my most common build, it builds the shared library so that downstream packages can use it, but produces a ~20% faster clang at the cost of disk space.